### PR TITLE
IA-1302 IA-1308 test new R kernel image

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-1302

superseded by https://github.com/DataBiosphere/leonardo/pull/1045
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins test

**Comment by Qi77Qi** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookRKernelSpec.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ import scala.language.postfixOps
   */
 @DoNotDiscover
 class NotebookRKernelSpec extends ClusterFixtureSpec with NotebookTestUtils {
-
+  override val jupyterDockerImage: Option[String] = Some("us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r:0.0.1")
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookRKernelSpec.scala]: Just curious if this should read from a conf file or something... when we update the R image we'll need to update this line. Not a blocker for this PR.

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala [hunk 2]
```diff
@@ -252,6 +248,10 @@ class ClusterMonitorActor(val cluster: Cluster,
         dataAccess.clusterQuery.completeDeletion(cluster.id)
       }
       _ <- authProvider.notifyClusterDeleted(cluster.internalId, cluster.auditInfo.creator, cluster.auditInfo.creator, cluster.googleProject, cluster.clusterName)
+      // Remove the Dataproc Worker IAM role for the cluster service account.
+      // Only happens if the cluster was created with a service account other
+      // than the compute engine default service account.
+      _ <- removeIamRolesForUser
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala]: Just reading the code, I'm not sure we need the `removeIamRolesForUser` method anymore. Can we get rid of `removeIamRolesForUser` and just call `clusterHelper.removeClusterIamRoles()` directly here?

The existing code did some checking to make sure we don't prematurely remove roles if the user has some Creating clusters. I don't think we need that if doing it at deletion time -- ClusterHelper should take care of all the logic.

**Review by rtitle — CHANGES_REQUESTED** [review]
Just 1 request for the IAM stuff, see above.

Also it seems a test failed: `NotebookRKernelSpec should be able to install mlr`

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookRKernelSpec.scala (hunk 1 above)
Comment (Qi77Qi) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookRKernelSpec.scala]: yea, I was just being lazy especially build has worked yet :)

**Comment by Qi77Qi** [comment]
clusterMonitorSpec seems to pass locally..jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Review by rtitle — APPROVED** [review]
This looks good, odd tests are failing. I'm actually deploying this branch to a fiab to see if the preemptibles error is fixed.

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
@Qi77Qi in my fiab (running Terra) I am able to pause/resume clusters with preemptibles, and also add/remove preemptibles on a cluster. It _feels_ like the issue is fixed. Let's see how this test run does.

**Comment by rtitle** [comment]
It failed again... I think I have a theory why. Gonna try 1 more change.

**Comment by rtitle** [comment]
I broke out my idea into a new PR: https://github.com/DataBiosphere/leonardo/pull/1045/

Haven't done much testing yet, will iterate.

**Pull request closed without merging**